File: examples/test_TS_and_Python_interoperability/encrypter.py
# encrypter.py
import base64
from Crypto.PublicKey import RSA
from Crypto.Cipher import PKCS1_OAEP
from Crypto.Hash import SHA1

# Generate a new RSA key pair
key = RSA.generate(2048)
private_key = key.export_key()
public_key = key.publickey().export_key()

# Save keys to files
with open("py_public_key.pem", "wb") as f:
    f.write(public_key)

with open("py_private_key.pem", "wb") as f:
    f.write(private_key)

password = "helloWorld"

def get_encrypted(challenge):
    # Extract timestamp as string
    timestamp = challenge.split("|")[1]
    pw_pair = password + "||" + timestamp
    
    # Load the public key
    public_key_obj = RSA.import_key(public_key)
    
    # Create cipher using PKCS1_OAEP
    cipher = PKCS1_OAEP.new(public_key_obj, hashAlgo=SHA1)
    
    # Encrypt the data
    encrypted_data = cipher.encrypt(pw_pair.encode('utf-8'))
    
    # Convert to base64
    return base64.b64encode(encrypted_data).decode()

challenge = "ezwXceQ63fV9oWTSJBAE2Zq1Cw5tBIJe+7+Rl8jrgbk=|1475429754114|4017bda8-0a15-4154-a8b7-88069b05cb4e"
encrypted = get_encrypted(challenge)
print("Encrypted data:", encrypted)

# Save encrypted data to file for TypeScript to read
with open("py_encrypted.txt", "w") as f:
    f.write(encrypted)

File: examples/test_TS_and_Python_interoperability/encrypter.ts
// encrypter.ts
import crypto from "crypto";
import fs from "fs";
import path from "path";

// Generate key pair
const { publicKey, privateKey } = crypto.generateKeyPairSync("rsa", {
  modulusLength: 2048,
  publicKeyEncoding: {
    type: "spki",
    format: "pem",
  },
  privateKeyEncoding: {
    type: "pkcs8",
    format: "pem",
  },
});

// Save keys to files
fs.writeFileSync("public_key.pem", publicKey);
fs.writeFileSync("private_key.pem", privateKey);

const password = "helloWorld";

function getEncrypted(challenge: string): string {
  const timestamp = challenge?.split("|")[1];
  const pw_pair = password + "||" + timestamp;
  const padding = crypto.constants.RSA_PKCS1_OAEP_PADDING;
  const encryptedData = crypto.publicEncrypt({ key: publicKey, padding }, Buffer.from(pw_pair)).toString("base64");
  return encryptedData;
}

const challenge = "ezwXceQ63fV9oWTSJBAE2Zq1Cw5tBIJe+7+Rl8jrgbk=|1475429754114|4017bda8-0a15-4154-a8b7-88069b05cb4e";
const encrypted = getEncrypted(challenge);
console.log("Encrypted data:", encrypted);

// Save encrypted data to file for Python to read
fs.writeFileSync("encrypted.txt", encrypted);
